p2p/background/router.go

```diff
@@ -342,13 +349,30 @@ func (rtr *backgroundRouter) bootstrap(ctx context.Context) error {
 			return nil
 		}
 
-		if err := rtr.host.Connect(ctx, libp2pAddrInfo); err != nil {
+		if err := rtr.connectWithRetry(ctx, libp2pAddrInfo); err != nil {
 			return fmt.Errorf("connecting to peer: %w", err)
 		}
 	}
 	return nil
 }
 
+// connectWithRetry attempts to connect to the given peer, retrying up to connectMaxRetries times
+// and waiting connectRetryTimeout between each attempt.
+func (rtr *backgroundRouter) connectWithRetry(ctx context.Context, libp2pAddrInfo libp2pPeer.AddrInfo) error {
```

Comment:
Nit: Is it worth considering exponential back-off here?

Reply:
https://github.com/pokt-network/pocket/pull/902/files/37b2e5cd4a0035fd64737ec01be36b0170dcd903#r1261840776